File: cpp/example_code/iam/iam_delete_access_key.cpp
/*
   Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.

   This file is licensed under the Apache License, Version 2.0 (the "License").
   You may not use this file except in compliance with the License. A copy of
   the License is located at

    http://aws.amazon.com/apache2.0/

   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
   CONDITIONS OF ANY KIND, either express or implied. See the License for the
   specific language governing permissions and limitations under the License.
*/
#include <aws/core/Aws.h>

#include <aws/iam/IAMClient.h>
#include <aws/iam/model/DeleteAccessKeyRequest.h>

#include <iostream>

/**
 * Deletes an access key from an IAM user, based on command line input
 */
int main(int argc, char** argv)
{
    if(argc != 3)
    {
        std::cout << "Usage: iam_delete_access_key <user_name> <access_key_id>" << std::endl;
        return 1;
    }

    Aws::String userName(argv[1]);
    Aws::String accessKeyId(argv[2]);

    Aws::SDKOptions options;
    Aws::InitAPI(options);

    Aws::IAM::IAMClient iamClient;

    Aws::IAM::Model::DeleteAccessKeyRequest deleteAccessKeyRequest;
    deleteAccessKeyRequest.SetUserName(userName);
    deleteAccessKeyRequest.SetAccessKeyId(accessKeyId);

    auto deleteAccessKeyOutcome = iamClient.DeleteAccessKey(deleteAccessKeyRequest);
    if(!deleteAccessKeyOutcome.IsSuccess())
    {
        std::cout << "Error deleting access key " << accessKeyId << " from user " << userName << ": " << deleteAccessKeyOutcome.GetError().GetMessage() << std::endl;
    }
    else
    {
        std::cout << "Successfully deleted access key " << accessKeyId << " for IAM user " << userName << std::endl;
    }

    Aws::ShutdownAPI(options);

    return 0;
}

File: javascript/example_code/iam/iam_deleteaccesskey.js
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

// ABOUT THIS NODE.JS SAMPLE: This sample is part of the SDK for JavaScript Developer Guide topic at
// https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/iam-examples-managing-access-keys.html

// snippet-start:[iam.JavaScript.keys.deleteAccessKey]
// Load the AWS SDK for Node.js
var AWS = require("aws-sdk");
// Set the region
AWS.config.update({ region: "REGION" });

// Create the IAM service object
var iam = new AWS.IAM({ apiVersion: "2010-05-08" });

var params = {
  AccessKeyId: "ACCESS_KEY_ID",
  UserName: "USER_NAME",
};

iam.deleteAccessKey(params, function (err, data) {
  if (err) {
    console.log("Error", err);
  } else {
    console.log("Success", data);
  }
});
// snippet-end:[iam.JavaScript.keys.deleteAccessKey]
